### PR TITLE
[Merged by Bors] - feat: port Order.Category.BddLatCat

### PR DESCRIPTION
**Pull request by joneugster**
---

This still contains the commits from the forward-port PR mentioned below.

- [x] depends on: #4990
- [x] depends on: #3918

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4990
* leanprover-community/mathlib4#3918
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/Category/BddLatCat.lean [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  comp_id := BoundedLatticeHom.id_comp
+  assoc _ _ _ := BoundedLatticeHom.comp_assoc _ _ _
+
+-- Porting note: added.
+instance instFunLike (X Y : BddLatCat) : FunLike (X ⟶ Y) X (fun _ => Y) :=
+  show FunLike (BoundedLatticeHom X Y) X (fun _ => Y) from inferInstance
+
+instance : ConcreteCategory BddLatCat where
+  forget := {
+    obj := (↑)
```
Comment (semorrison) [inline Mathlib/Order/Category/BddLatCat.lean:75]:
```suggestion
  forget :=
  { obj := (↑)
```

File: Mathlib/Order/Category/BddLatCat.lean [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+          | some none => rfl
+          | some (some a) => rfl
+      homEquiv_naturality_right := fun f g => LatticeHom.ext fun a => rfl }
+#align Lat_to_BddLat_forget_adjunction latToBddLatCatForgetAdjunction
+
+
+/-- `latToBddLatCat` and `OrderDual` commute. -/
+-- Porting note: There is a warning plus `simpNF` linter complaining which I don't feel like
+-- debugging at the moment.
+-- @[simps]
```
Comment (semorrison) [inline Mathlib/Order/Category/BddLatCat.lean:233]: The message is just saying that you need to switch to `@[simps!]`. I've pushed this change.

**Comment by semorrison** [comment]
bors d+

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4990~~
* ~~leanprover-community/mathlib4#3918~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/Category/BddLatCat.lean (hunk 2 above)
Comment (joneugster) [reply Mathlib/Order/Category/BddLatCat.lean]: That porting note wasn't that clear: the part I didn't understand was the `simpNF` error:

```
Left-hand side simplifies from
  ↑(latToBddLatCatCompDualIsoDualCompLatToBddLatCat.hom.app X).toSupHom a✝
to
  ↑(latToBddLatCatCompDualIsoDualCompLatToBddLatCat.hom.app X).toSupHom a✝
```

but I'll look into it

Comment (joneugster) [reply Mathlib/Order/Category/BddLatCat.lean]: Nah I got nothing :(

@semorrison would you like to have this investigated or is `nolint simpNF` okay?

Comment (joneugster) [reply Mathlib/Order/Category/BddLatCat.lean]: `nolint simpNF` is not enough. I went back to uncommenting `@[simps!]`. If somebody wants to investigate before this is merged, feel free!

**Comment by joneugster** [comment]
I just finished porting PRs for all files depending on this, and the `simps` issue did not have any impact. Therefore I think I'll just kick this on bors and we can address the `porting note` later if desired.

bors r+

**Pull request closed without merging**